### PR TITLE
eth2util/keystore: concurrently read/write keystore files

### PR DESCRIPTION
**Pull request by gsora**
Miga Labs testing revealed that loading a huge amount of keys (1000 in their case) takes an unbearable amount of time.

Manual debugging with a similar amount of secure keys also revealed that writing them is slow.

Use `errgroup` to make the reading/writing process concurrent, utilising as much host resources as possible, reducing the time needed to execute those operations.

category: feature
ticket: none

### CODE REVIEW

File: eth2util/keystore/keystore.go [hunk 1]
```diff
@@ -105,29 +119,43 @@ func loadFiles(dir string, sortKeyfiles func([]string) ([]string, error)) ([]tbl
 		}
 	}
 
-	var resp []tbls.PrivateKey
-	for _, f := range files {
-		b, err := os.ReadFile(f)
-		if err != nil {
-			return nil, errors.Wrap(err, "read file")
-		}
+	eg := errgroup.Group{}
```
[the remark this answers is not in the document]
Comment (corverroos) [reply eth2util/keystore/keystore.go:122]: as per https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_use_a_consistent_declaration_style

**Review by corverroos — CHANGES_REQUESTED** [review]

File: eth2util/keystore/keystore.go [hunk 2]
```diff
@@ -54,32 +55,45 @@ func StoreKeys(secrets []tbls.PrivateKey, dir string) error {
...
+			if err := storePassword(filename, password); err != nil {
+				return errors.Wrap(err, "store password", z.Str("filename", filename))
+			}
+
+			return nil
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return errors.Wrap(eg.Wait(), "store keys")
```
[the remark this answers is not in the document]
Comment (gsora) [reply eth2util/keystore/keystore.go:96]: oof, copypaste error

File: eth2util/keystore/keystore.go [hunk 3]
```diff
@@ -105,29 +119,43 @@ func loadFiles(dir string, sortKeyfiles func([]string) ([]string, error)) ([]tbl
...
+			if err != nil {
+				return errors.Wrap(err, "load password", z.Str("filename", f))
+			}
+
+			secret, err := decrypt(store, password)
+			if err != nil {
+				return errors.Wrap(err, "keystore decryption", z.Str("filename", f))
+			}
+
+			resp[idx] = secret
```
[the remark this answers is not in the document]
Comment (gsora) [reply eth2util/keystore/keystore.go:151]: `resp` is an array, as long as you're writing to individual "slots" you can do that concurrently

[the remark this answers is not in the document]
Comment (corverroos) [reply eth2util/keystore/keystore.go:151]: ah, ok, maybe just add a comment to make this clear

**Review by corverroos — APPROVED** [review]

**Pull request merged**